### PR TITLE
Replace ovirtsdk with ovirtsdk4

### PR DESCRIPTION
**Pull request by chloerh**
Signed-off-by: Haijiao Zhao <haizhao@redhat.com>

### CODE REVIEW

**Comment by chloerh** [comment]
Test result of some cases
--------------

Test name | Duration | Status
-- | -- | --
esx.vm.5_5.linux.6_9.arch_i386.raw_f.NFS | 45 min | Fixed
esx.vm.5_5.windows.win8_1.arch_x86_64.raw_f.NFS | 51 min | Fixed
esx.vm.6_0.linux.6_9.arch_x86_64.raw_f.NFS | 27 min | Fixed
esx.vm.6_0.windows.win2012r2.arch_x86_64.raw_f.NFS | 28 min | Fixed
esx.vm.6_5.windows.win10.arch_x86_64.raw_f.NFS | 36 min | Fixed
esx.vm.6_7.linux.7_5.arch_x86_64.raw_f.NFS | 35 min | Fixed
esx.vm.6_7.windows.win2016.arch_x86_64.raw_f.NFS | 36 min | Fixed
kvm.linux.5_11.arch_i386.raw_f.NFS | 7 min 10 sec | Fixed
kvm.linux.6_9.arch_i386.raw_f.NFS | 9 min 34 sec | Fixed
kvm.linux.6_9.arch_x86_64.qcow2_f.NFS | 9 min 28 sec | Fixed
kvm.linux.7_5.arch_x86_64.qcow2_f.NFS | 10 min | Fixed
kvm.windows.win2016.arch_x86_64.qcow2_f.NFS | 20 min | Fixed
kvm.windows.win8_1.arch_i386.qcow2_f.NFS | 16 min | Fixed
xen.vm.hvm.linux.7_5.arch_x86_64.raw_f.NFS | 14 min | Fixed
xen.vm.hvm.windows.win2008r2.arch_x86_64.raw_f.NFS | 20 min | Fixed
xen.vm.pv.linux.6_9.arch_i386.raw_f.NFS | 12 min | Fixed

File: virttest/ovirt.py [hunk 1]
```diff
@@ -9,10 +9,10 @@
 import logging
 
 try:
-    from ovirtsdk.api import API
-    from ovirtsdk.xml import params as param
+    import ovirtsdk4 as sdk
```
Comment (chunfuwen) [inline virttest/ovirt.py]: suggest import different modules as per specific ovirt version, otherwise it will lose compatibility

Comment (chloerh) [reply virttest/ovirt.py]: We will use ovirt4 in both python2 and python3 env, therefore no conflict

File: virttest/ovirt.py [hunk 2]
```diff
@@ -322,7 +340,7 @@ def delete(self, timeout=300):
         end_time = time.time() + timeout
         if self.name in self.list():
             logging.info('Delete VM %s' % self.name)
-            self.instance.delete()
+            self.instance.remove()
```
[the remark this answers is not in the document]
Comment (chloerh) [reply virttest/ovirt.py]: I don't understand. This function is to delete the target vm from ovirt.

File: virttest/ovirt.py [hunk 3]
```diff
@@ -584,10 +614,11 @@ class DataCenterManager(object):
     def __init__(self, params):
         self.name = params.get("dc_name", "")
         self.params = params
-        (self.api, self.version) = connect(params)
+        (self.connection, self.version) = connect(params)
+        self.dcs_service = self.connection.system_service().data_centers_service()
 
         if self.name:
-            self.instance = self.api.datacenters.get(self.name)
+            self.instance = self.dcs_service.list(search='name=%s' % self.name)[0]
```
[the remark this answers is not in the document]
Comment (chloerh) [reply virttest/ovirt.py]: Updated.

File: virttest/ovirt.py (hunk 1 above)
Comment (xiaodwan) [reply virttest/ovirt.py]: rhv3.6 is going to out of service, it's not necessary to be compatible with an out of service product.

File: virttest/ovirt.py [hunk 4]
```diff
@@ -58,21 +58,26 @@ def connect(params):
...
         if not _connected:
-            _api = API(url, username, password, insecure=True)
+            connection = sdk.Connection(
+                url=url,
+                username=username,
+                password=password,
+                insecure=True
+            )
             _connected = True
-            return (_api, version)
```
Comment (chunfuwen) [inline virttest/ovirt.py]: In both cases, they return same result:connection, version

File: virttest/ovirt.py [hunk 5]
```diff
@@ -352,7 +370,11 @@ def delete_from_export_domain(self, export_name):
         """
         vm = self.lookup_by_storagedomains(export_name)
         try:
-            vm.delete()
+            sds_service = self.connection.system_service().storage_domains_service()
+            export_sd = sds_service.list(search=export_name)[0]
+            logging.info('Remove VM %s from export storage' % self.name)
+            export_vms_service = sds_service.storage_domain_service(export_sd.id).vms_service()
+            export_vms_service.vm_service(vm.id).remove()
```
Comment (chunfuwen) [inline virttest/ovirt.py]: those code are duplicated to previous, can we wrap them into one function instead?

**Pull request merged**